documentation/smg/08_setting_up_jenkins_ci.md

```diff
@@ -0,0 +1,126 @@
+# Setting Up Jenkins CI
+
+We use machines administered by [ANVIL](https://anvil.softeng-support.ac.uk/)
+to run our continuous integration (CI) jobs. These machines are running
+Jenkins v2.190.3.
+
+We would like to be able to script and commit as much of our CI configuration
+as possible. This makes things re-creatable and locally testable. However,
+there is necessarily some set up that must be done within the GUI in Jenkins.
+It is therefore important that this document is kept up-to-date with the steps
+taken to reach the configuration we use.
+
+We will use the CI for building pull requests and building master every
+evening. Each build will create a Horace package that is shippable to users.
+
+## Overview
+
+Our CI builds follow the following high-level process:
+
+- Jenkins checks out the correct branch
+
+  For pull requests:
+
+  - Webhooks are created in GitHub that notify Jenkins when a pull request event
+  occurs.
+  - When Jenkins receives a notification from GitHub it will checkout out the
+  relevant pull request branch and merge it with master.
+
+  For nightly builds:
+
+  - Jenkins will clone master at a specific time each evening.
+
+- Jenkins loads and runs the [`Jenkinsfile`](./../../tools/build_config/Jenkinsfile)
+located in `tools/build_config`.
+- The `Jenkinsfile` script runs our build script (one for Windows, one for Unix)
+and updates GitHub about the status of the build.
+
+## More Detailed Set Up
+
+### GitHub
+
+The only job required on GitHub is to set up webhooks to notify Jenkins of pull
+requests. These can be created by GitHub repository admins by opening the
+settings tab in the [main repo](https://github.com/pace-neutrons/Horace).
+
+<img src="./images/08_github_settings.png">
+
+Then selecting the `Webhooks` menu item on the left-hand side.
+
+Your webhook should have payload URL
+`https://anvil.softeng-support.ac.uk/jenkins/generic-webhook-trigger/invoke?token=<my_secret_token>`
+, content type `application/json` and only trigger on pull request events. This
+will send a json containing information about the pull request (including pull
+request number and the action taken in the pull request event) to Jenkins. Pull
+request events trigger when a pull request is:
+
+> *opened, closed, reopened, edited, assigned, unassigned, review requested,
+> review request removed, labeled, unlabeled, synchronized, ready for review,
+> locked, or unlocked.*
+
+This means that, from the webhook's JSON, you must decide if you want to trigger
+a build. usually you'd only want to trigger builds when a pull request is
+`opened` or `synchronized`.
```

Comment:
"`syncrhonied` (fired when a new commit is pushed to the PR)" -- otherwise it's not obvious what causes the sync event